### PR TITLE
Implement spmatrix that is a base class of sparse matrixes

### PR DESCRIPTION
**Pull request by unnonouno**
I implemented spmatrix that is a base class of all other sparse matrixes. It is compatible with `scipy.sparse.spmatrix`.
related to #36 
merge #39 before this PR.

### CODE REVIEW

File: tests/cupy_tests/sparse_tests/test_base.py [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+        s = dummies[sp](shape=(1, 1), nnz=1)
+        return bool(s)
+
+    @testing.numpy_cupy_equal(sp_name='sp')
+    def test_bool_false(self, xp, sp):
+        s = dummies[sp](shape=(1, 1), nnz=0)
+        return bool(s)
+
+    @testing.numpy_cupy_raises(sp_name='sp', accept_error=ValueError)
+    def test_bool_true(self, xp, sp):
```
Comment (okuta) [inline tests/cupy_tests/sparse_tests/test_base.py]: There are two `test_bool_true`. Please rename.

**Comment by okuta** [comment]
Please add `scipy` to test dependency.

**Comment by unnonouno** [comment]
Ah, scipy support is optional. So, i'll make the test optional.

**Comment by okuta** [comment]
LGTM!

And, please send a PR to chaienr-test.

**Pull request merged**